app/src/main/java/dev/arkbuilders/arkmemo/data/repositories/NotesRepoImpl.kt

```diff
@@ -0,0 +1,87 @@
+package dev.arkbuilders.arkmemo.data.repositories
+
+import android.util.Log
+import dev.arkbuilders.arklib.ResourceId
+import dev.arkbuilders.arklib.data.index.RootIndex
+import dev.arkbuilders.arklib.user.properties.Properties
+import dev.arkbuilders.arklib.user.properties.PropertiesStorage
+import dev.arkbuilders.arklib.user.properties.PropertiesStorageRepo
+import dev.arkbuilders.arkmemo.data.ResourceMeta
+import dev.arkbuilders.arkmemo.models.BaseNote
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+import kotlinx.coroutines.withContext
+import java.nio.file.Path
+import javax.inject.Inject
+import kotlin.io.path.deleteIfExists
+import kotlin.io.path.extension
+import kotlin.io.path.getLastModifiedTime
+import kotlin.io.path.moveTo
+import kotlin.io.path.name
+
+abstract class NotesRepoImpl<Note> constructor(
+    protected val root: Path,
```

Comment:
Why do you need this protected? Name of NotesRepoImpl states that it is an implementation, please make it to be normal class, not abstract. And make sure there is not class extends from this class

Reply:
This class contains functions and properties common to all repos. I extracted them from TextNotesRepo and GraphicNotesRepo. Should we put them in the interface?